CI: Install and use lpython as a package

Shaikh-Ubaid opened the pull request:
Using `pip install lpython_emulation`, we can now install and import `lpython` as a package. This PR updates the CI to use `lpython` installed via `pip` (previously we used `lpython` from the runtime library).

Shaikh-Ubaid commented:
I updated the description of this PR. This is ready.

certik commented on .github/workflows/CI.yml:
```diff
@@ -225,6 +225,8 @@ jobs:
         run: |
+            # Package lpynn has lpython_emulation as dependency
+            # Hence, it should by default install lpython_emulation
             python -m pip install lpynn numpy
```
I think that this is problematic. If we change anything in `lpython.py` this test will break, since we have to update it in `lpynn` as well.

At the very least, let's pin a specific version of the lpynn package, as well as numpy.

Shaikh-Ubaid replied:
Done.

Shaikh-Ubaid commented:
We updated the `lnn` package in https://github.com/lcompilers/lpython/pull/1751. I missed to reflect/propagate the updates to `integration_tests/test_pip_import_01.py` and `lpynn`. I updated them in this PR.

certik reviewed (approved):
Awesome. I think for now this is good. Thanks!
